File: server/src/controllers/paymentController.js
const receiptService = require('../services/receiptService');
const { asyncHandler } = require('../middleware/errorHandler');
const logger = require('../lib/logger');

const processPayment = asyncHandler(async (req, res) => {
    const { rideId, amount } = req.body;

    // Mock payment success
    const transactionId = 'txn_' + Math.random().toString(36).substr(2, 9);

    logger.info('Payment processed', { rideId, amount, transactionId });

    // Update receipt with payment status
    try {
        await receiptService.updateReceiptPaymentStatus(rideId, 'completed', transactionId);
    } catch (error) {
        logger.warn('Receipt not found for payment update', { rideId });
    }

    res.json({
        success: true,
        transactionId,
        amount,
        rideId
    });
});

module.exports = { processPayment };

File: server-java/src/main/java/com/ridehailing/controller/PaymentController.java
package com.ridehailing.controller;

import com.ridehailing.dto.ProcessPaymentRequest;
import com.ridehailing.service.ReceiptService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;
import java.util.UUID;

@RestController
@RequestMapping("/v1")
@RequiredArgsConstructor
@Slf4j
public class PaymentController {

    private final ReceiptService receiptService;

    /**
     * Process payment for a ride
     * POST /v1/payments
     */
    @PostMapping("/payments")
    public ResponseEntity<Map<String, Object>> processPayment(@Valid @RequestBody ProcessPaymentRequest request) {
        // Mock payment - in production, integrate with Stripe/PayPal
        String transactionId = "txn_" + UUID.randomUUID().toString().substring(0, 9);

        log.info("Payment processed: rideId={}, amount={}, transactionId={}",
                request.getRideId(), request.getAmount(), transactionId);

        // Update receipt with payment status
        try {
            receiptService.updateReceiptPaymentStatus(request.getRideId(), "completed", transactionId);
        } catch (Exception e) {
            log.warn("Receipt not found for payment update: rideId={}", request.getRideId());
        }

        return ResponseEntity.ok(Map.of(
                "success", true,
                "transactionId", transactionId,
                "amount", request.getAmount(),
                "rideId", request.getRideId()));
    }
}
